modules/opts.nix

```diff
@@ -36,15 +36,29 @@ let
   };
 in
 {
-  options = lib.mapAttrs (
-    _:
-    { description, ... }:
-    lib.mkOption {
-      type = with lib.types; attrsOf anything;
-      default = { };
-      inherit description;
-    }
-  ) optionsAttrs;
+  options =
+    (lib.mapAttrs (
+      _:
+      { description, ... }:
+      lib.mkOption {
+        type = with lib.types; attrsOf anything;
+        default = { };
+        inherit description;
+      }
+    ) optionsAttrs)
+    // {
+      globalsPre = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        internal = true;
+      };
+
+      globalsPost = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        internal = true;
+      };
+    };
```

Comment:
I guess the intention is that this is lua code injected pre/post the lua impl for the `globals` option.

But reading the option name I'd half-expect to be able to assign `globals` earlier or later than normal.

This feels like working around a poor design rather than an objective improvement.

Reply:
I'm wondering if instead, each plugin should be responsible for using `vim.g[k] = ` itself. Perhaps within `luaConfig.init` ?

Yes, this would mean vim-plugins aren't taking advantage of the `globals` option, and consiquently end-users won't see the configured globals when inspecting the option's value. But perhaps that is worth it in order to locate the globals in a more predictable location?

Reply:
Another idea: Perhaps the `luaConfig` submodule could also be used for non-plugin modules too? E.g. we could have a `globalsLuaConfig` option, and move the lua impl to `globalsLuaConfig.init`?

_(maybe rename `init` something like `main`, `content`, `text`, etc?)_

Reply:
I'm not really a fan of any of the three solutions, tbh. So open to alternative proposals too.

Reply:
> I'm wondering if instead, each plugin should be responsible for using `vim.g[k] = ` itself. Perhaps within `luaConfig.init` ?
> 
> Yes, this would mean vim-plugins aren't taking advantage of the `globals` option, and consiquently end-users won't see the configured globals when inspecting the option's value. But perhaps that is worth it in order to locate the globals in a more predictable location?

I though about that, but I think setting the global options in nix is a bit better, as it allows to do more compile time stuff (though you could _still_ do compile time stuff by accessing the plugin's settings)

Reply:
> Another idea: Perhaps the `luaConfig` submodule could also be used for non-plugin modules too? E.g. we could have a `globalsLuaConfig` option, and move the lua impl to `globalsLuaConfig.init`?
> 
> _(maybe rename `init` something like `main`, `content`, `text`, etc?)_

There _could_ be a use for it, for example to define helpers for keymaps, or options